website/src/app/kb/architecture/critical-sequences/readme.mdx

```diff
@@ -0,0 +1,190 @@
+import SupportOptions from "@/components/SupportOptions";
+import NextStep from "@/components/NextStep";
+import Alert from "@/components/DocsAlert";
+import Link from "next/link";
+import Image from "next/image";
+
+# Architecture: Critical Sequences
+
+Firezone is a distributed system with many moving parts, but some parts are
+especially critical to the integrity of the entire system:
+
+- [Authentication](#authentication): User authentication, usually with an
+  identity provider.
+- [Policy evaluation](#policy-evaluation): Deciding whether to allow or deny a
+  connection request.
+- [DNS resolution](#dns-resolution): Resolving DNS-based Resources.
+
+These will be explained in more detail below.
+
+## Authentication
+
+Firezone authenticates users using two primary methods:
+
+- [Email (OTP)](/kb/authenticate/email): Users receive a one-time password (OTP)
+  via email.
+- [OpenID Connect](/kb/authenticate/oidc): Users authenticate with an identity
+  provider that supports
+  [OpenID Connect (OIDC)](https://openid.net/developers/how-connect-works/).
+
+The authentication process for each is similar. Both methods begin the
+authentication process at your Firezone account's sign in page:
+`https://app.firezone.dev/<your-account>`.
+
+However, the OIDC flow redirects the user to the identity provider for
+authentication before the final redirect back to Firezone.
+
+Here's how the authentication flow works:
+
+<Link
+  target="_blank"
+  href="/images/kb/architecture/critical-sequences/authentication.svg"
+>
+  <Image
+    src="/images/kb/architecture/critical-sequences/authentication.svg"
+    alt="Firezone authentication sequence diagram"
+    width={1200}
+    height={1200}
+  />
+</Link>
+
+1. User clicks `Sign in` from the Client.
+1. The Client generates random 32-byte `state` and `nonce` values. These are
+   used to prevent certain kinds of forgery and injection attacks.
+1. A browser window opens to your account's sign in page,
+   `https://app.firezone.dev/<your-account>` containing the `nonce` and `state`
+   parameters.
+1. The user chooses which authentication method to use. If OIDC, the user is
+   redirected out to the identity provider.
+1. After successfully authenticating, the user is redirected back to the admin
+   portal.
+1. The admin portal mints a Firezone `token` created from the `nonce` parameter
+   and other information.
+1. The admin portal issues a final redirect to
+   `firezone-fd0020211111://handle_client_sign_in_callback` with the `token` and
```

Comment:
Not sure if you want parameter names to be accurate here or not, but we return `fragment`:
```suggestion
   `firezone-fd0020211111://handle_client_sign_in_callback` with the `fragment` and
```

Reply:
The goal here was to get the basic flow across and not necessarily describe an API -- I worry this is too much detail for semi-technical folks to read (or technical but who just want an overview)

Reply:
This might be better as code comments we can link to or on a new `security architecture` page after pentest or something